src/syscall/forklock.go

```diff
@@ -0,0 +1,38 @@
+//go:build tinygo && linux && !wasip1 && !wasip2 && !darwin && !baremetal
+
+package syscall
+
+// This is the original ForkLock:
+//
+// var ForkLock sync.RWMutex
+//
+// This requires importing sync, but importing sync causes an import loop:
+//
+// package tinygo.org/x/drivers/examples/net/tcpclient
+//        imports bytes
+//        imports io
+//        imports sync
+//        imports internal/task
+//        imports runtime/interrupt
+//        imports device/arm
+//        imports syscall
+//        imports sync: import cycle not allowed
```

Comment:
This shouldn't be happening: you have excluded the `baremetal` build tag so `device/arm` should not be imported.

Reply:
Sounds reasonable. This is leftover code from the old net rework PR I based this one on, I dragged it with me and forgot to remove it.